core/admin/admin_interface.go

```diff
@@ -0,0 +1,98 @@
+package admin
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var (
+	AdminContractAddr = common.HexToAddress("0x010000000000000000000000000000000000000a")
+)
+
+// StateDB is the interface for accessing EVM state
+type StateDB interface {
+	GetState(common.Address, common.Hash) common.Hash
+	// SetState(common.Address, common.Hash, common.Hash)
+
+	// SetCode(common.Address, []byte)
+
+	// SetNonce(common.Address, uint64)
+	// GetNonce(common.Address) uint64
+
+	// GetBalance(common.Address) *big.Int
+	// AddBalance(common.Address, *big.Int)
+	// SubBalance(common.Address, *big.Int)
+
+	// CreateAccount(common.Address)
+	// Exist(common.Address) bool
+
+	// Suicide(common.Address) bool
+	// Finalise(deleteEmptyObjects bool)
+}
+
+type KYCStatus int
+
+const (
+	KYC_UNKNOWN  = -1
+	KYC_VERIFIED = 1
+	KYC_EXPIRED  = 2
+
+	FucntionSelectorLength = 4
+)
+
+var (
+	DisableContractFunctionSelector = common.FromHex("0x00000000")
+)
+
+func KycRoleKeyFromAddr(addr common.Address) common.Hash {
+	return crypto.Keccak256Hash(append(addr.Hash().Bytes(), common.HexToHash("0x2").Bytes()... /*slot 2 reference admin.sol map(address => uint)*/))
```

Comment:
```suggestion
	return crypto.Keccak256Hash(addr.Hash().Bytes(), common.HexToHash("0x2").Bytes()/*slot 2 reference admin.sol map(address => uint)*/)
```
you can drop `append` which leads the same result